### PR TITLE
feat: Implement CloudAsset crawler

### PR DESCRIPTION
**Pull request by shravankshenoy**
## Description

- Issue #247 Added support for CloudAsset resource 

## Changes Made
- Created client factory for CloudAsset resource : `cloudasset_client.py`
- Created a new crawler factory in the crawler package : `cloud_asset_crawler.py`
- In the scanner.py file, included the crawler in the crawl loop and updated the crawl_client_map. Also modified example.config
- Added unit tests in `test_unit.py`

## Testing
App ran successfully during local testing

### CODE REVIEW

File: src/gcp_scanner/test_unit.py [hunk 1]
```diff
@@ -612,6 +614,22 @@ def test_pubsub_subs(self):
       )
     )
 
+  def test_cloud_assets(self):
+    """Test CloudAsset list"""
+    self.assertTrue(
+      verify(
```
Comment (mshudrak) [inline src/gcp_scanner/test_unit.py:620]: So, the test is failing right now which is expected. Could you please create an empty cloud_assets file similar to other crawlers. I will populate it with actual data for testing afterwards.

**Review by mshudrak — CHANGES_REQUESTED** [review]
Thanks for PR. Let's start with making the tests engine happy.

File: src/gcp_scanner/test_unit.py (hunk 1 above)
Comment (shravankshenoy) [reply src/gcp_scanner/test_unit.py:620]: Oops, missed that. I ran pytest without the file and got the error below (as expected)
image

I have made a commit to add the empty file. Now that error is no longer appearing.

**Comment by shravankshenoy** [comment]
This is part of the output generated when I run the CloudAsset crawler. The actual list is much bigger, because of the the huge number of asset types Google Cloud supports.

Found complete list of supported asset types in the following link : https://cloud.google.com/asset-inventory/docs/supported-asset-types

```

{ 
"cloud_assets": [
     {
      "name": "//cloudbilling.googleapis.com/projects/circular-fusion-406401/billingInfo",
      "assetType": "cloudbilling.googleapis.com/ProjectBillingInfo",
      "ancestors": [
        "projects/674352937902"
      ],
      "updateTime": "2023-11-27T01:41:08.364780Z"
    },
    {
      "name": "//cloudresourcemanager.googleapis.com/projects/674352937902",
      "assetType": "cloudresourcemanager.googleapis.com/Project",
      "ancestors": [
        "projects/674352937902"
      ],
      "updateTime": "2023-11-27T01:41:03.707782Z"
    },
    {
      "name": "//logging.googleapis.com/projects/674352937902/locations/global/buckets/_Default",
      "assetType": "logging.googleapis.com/LogBucket",
      "ancestors": [
        "projects/674352937902"
      ],
      "updateTime": "2023-11-27T06:54:40.032022Z"
    },
    {
      "name": "//serviceusage.googleapis.com/projects/674352937902/services/bigquery.googleapis.com",
      "assetType": "serviceusage.googleapis.com/Service",
      "ancestors": [
        "projects/674352937902"
      ],
      "updateTime": "2023-11-27T13:04:03.814827Z"
    },
    {
      "name": "//serviceusage.googleapis.com/projects/674352937902/services/bigquerymigration.googleapis.com",
      "assetType": "serviceusage.googleapis.com/Service",
      "ancestors": [
        "projects/674352937902"
      ],
      "updateTime": "2023-11-27T13:04:03.814827Z"
    },
    {
      "name": "//serviceusage.googleapis.com/projects/674352937902/services/bigquerystorage.googleapis.com",
      "assetType": "serviceusage.googleapis.com/Service",
      "ancestors": [
        "projects/674352937902"
      ],
      "updateTime": "2023-11-27T13:04:03.814827Z"
    },
    {
      "name": "//serviceusage.googleapis.com/projects/674352937902/services/cloudapis.googleapis.com",
      "assetType": "serviceusage.googleapis.com/Service",
      "ancestors": [
        "projects/674352937902"
      ],
      "updateTime": "2023-11-27T13:04:03.814827Z"
    },
    {
      "name": "//serviceusage.googleapis.com/projects/674352937902/services/cloudasset.googleapis.com",
      "assetType": "serviceusage.googleapis.com/Service",
      "ancestors": [
        "projects/674352937902"
      ],
      "updateTime": "2023-11-27T13:04:03.814827Z"
    }
]
}
```

File: src/gcp_scanner/test_unit.py (hunk 1 above)
Comment (mshudrak) [reply src/gcp_scanner/test_unit.py:620]: Ok, I added what's necessary in that file for the check.

Comment (mshudrak) [reply src/gcp_scanner/test_unit.py:622]: The test is still failing. Not sure why.

Comment (aryanagrawal22) [reply src/gcp_scanner/test_unit.py:622]: Yes, I think the tests would work fine without the `cloud_assets` key and just the list

Comment (shravankshenoy) [reply src/gcp_scanner/test_unit.py:622]: Hi @mshudrak , I updated the `cloud_assets` test file, following which the test passed successfully for me locally. I have commited the change to this branch. Could you rerun the tests and see if they are passing now (for cloud assets)?

Comment (mshudrak) [reply src/gcp_scanner/test_unit.py:622]: Thanks for the new change but it is still failing for some reason the output is empty. I tried to debug but do not see any obvious reason of why this is happenning.

Comment (shravankshenoy) [reply src/gcp_scanner/test_unit.py:622]: I see. Maybe there is something happening under the hood which we might be missing. I will try to do some research on my side as well, and share if I get any info. Will close this PR for the time being (and open it in the future if any solution is found). Hope that works :)

**Pull request closed without merging**